### PR TITLE
Implement share web api with css

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/0/1205583865502218/f 

### Description
This PR adds support for the share web API through c-s-s

### Steps to test this PR

- [x] Use `https://jsonblob.com/api/1169732359619993600` as the remote config
- [x] Go to `https://mdn.github.io/dom-examples/web-share/`
- [x] Click on Share MDN
- [x] A share sheet should show up 
- [x] Dismiss it
- [x] Web should say Error: DataError: User aborted
- [x] Click again
- [x] Copy to clipboard 
- [x] Web should say MDN shared successfully

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205966966963452

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3870** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3870?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3870?utm_source=stack-comment).

File: app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        context: Context,
+        input: JsCallbackData,
+    ): Intent {
+        data = input
+        val url = runCatching { input.params.getString("url") }.getOrNull().orEmpty()
+        val text = runCatching { input.params.getString("text") }.getOrNull().orEmpty()
+        val title = runCatching { input.params.getString("title") }.getOrNull().orEmpty()
+
+        val finalText = if (url.isNotEmpty() && text.isNotEmpty()) {
+            "$url $text"
```
Comment (muodov) [inline app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt:40]: this combination is already done in JS, url and text are never passed at the same time

File: app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            url
+        } else {
+            text
+        }
+
+        val finalTitle = if (title.isNotEmpty() && text.isNotEmpty()) {
+            ""
+        } else {
+            title
+        }
```
Comment (muodov) [inline app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt:51]: Should this be 
```suggestion
        val finalTitle = if (title.isNotEmpty()) {
            title
        } else {
            ""
        }
```

File: app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+        val getContentIntent = Intent(Intent.ACTION_SEND).apply {
+            type = "text/html"
+            putExtra(Intent.EXTRA_TEXT, finalText)
+            putExtra(Intent.EXTRA_TITLE, finalTitle)
```
Comment (muodov) [inline app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt:56]: Would it make sense to not pass this at all if the title is empty or absent?

File: app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                id = data.id,
+                params = JSONObject("""{}"""),
+            )
+        }
+        if (resultCode == Activity.RESULT_CANCELED) {
+            return JsCallbackData(
+                featureName = data.featureName,
+                method = data.method,
+                id = data.id,
+                params = JSONObject("""{ "failure": {"name":"AbortError", "message":"User aborted"} }"""),
```
Comment (muodov) [inline app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt:80]: let's make it similar to Chrome
```suggestion
                params = JSONObject("""{ "failure": {"name":"AbortError", "message":"Share canceled"} }"""),
```

Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt:71]: Done!

File: app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt (hunk 1 above)
Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/browser/webshare/WebShareChooser.kt]: 👍

File: content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt [hunk 5]
```diff
@@ -54,31 +55,25 @@ class ContentScopeScriptsJsMessaging @Inject constructor(
...
-                        response?.let {
-                            jsMessageHelper.sendJsResponse(response, callbackName, secret, webView)
-                        }
-                    }
+                    }?.process(jsMessage, secret, webView, jsMessageCallback)
                 }
             }
         } catch (e: Exception) {
-            logcat { "Exception is ${e.message}" }
+            logcat { "Exception is ${e.stackTraceToString()}" }
```
Comment (aitorvs) [inline content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt:76]: nit: I'd change it to `logcat(LogPriority.ERROR) { "Exception is ${e.asLog()} }`

**Pull request merged**